### PR TITLE
helm: make istiod chart consistency with istioctl

### PR DESCRIPTION
**Pull request by ichbinfrog**
**Please provide a description of this PR:**

This PR enables validating webhook by default for the istiod helm chart.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
This should have caught by virtual service validation. I see this error
admission webhook "validation.istio.io" denied the request: configuration is invalid: empty domain name not allowed

Did you enable validation webhook?

**Comment by ichbinfrog** [comment]
My bad @ramaraochavali , I did not have the validation webhook enabled in our clusters. 

Despite disabling the validating webhook having high impact on the stability of istiod, it does not seem to be documented in the [istiod](https://github.com/istio/istio/tree/master/manifests/charts/istio-control/istio-discovery) helm chart nor set as a default value. If you don't mind, I'll open a PR to address it

**Comment by ramaraochavali** [comment]
updating documentation is fine but changing the default we need @istio/wg-environments-maintainers  opinion also

**Comment by zirain** [comment]
@ichbinfrog did you install istio via helm-chart?

**Comment by ichbinfrog** [comment]
> updating documentation is fine but changing the default we need @istio/wg-environments-maintainers opinion also

Agreed, then I'll switch this PR to that purpose.

**Comment by ichbinfrog** [comment]
> @ichbinfrog did you install istio via helm-chart?

Yes, istiod is installed via the helm chart version 1.17.2 with a minimal values file:
```yaml
global:
  variant: distroless
  defaultPodDisruptionBudget:
    enabled: true
  proxy: 
    excludeOutboundPorts: "5432,6379"
    resources:
      requests:
        cpu: "20m"
        memory: "64Mi"
```

**Comment by zirain** [comment]
> > @ichbinfrog did you install istio via helm-chart?
> 
> Yes, istiod is installed via the helm chart version 1.17.2 with a minimal values file:
> 
> ```yaml
> global:
>   variant: distroless
>   defaultPodDisruptionBudget:
>     enabled: true
>   proxy: 
>     excludeOutboundPorts: "5432,6379"
>     resources:
>       requests:
>         cpu: "20m"
>         memory: "64Mi"
> ```

I think it's reason to add `configValidation: true` in istiod chart's values file, that will keep consistency bewteen istioctl and helm chart installation.

would you like to send an PR for that?

**Comment by ichbinfrog** [comment]
Sure thing @zirain , I've updated this PR accordingly 🙏

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -217,6 +217,9 @@ global:
     #   cpu: 100m
     #   memory: 128Mi
 
+  # enable validating webhooks to ensure that CRDs created are valid
+  configValidation: true
```
Comment (hzxuzhonghu) [inline manifests/charts/istio-control/istio-discovery/values.yaml:221]: can we prohibit disabling validation at all

**Comment by howardjohn** [comment]
I put a hold since this looks sensitive and I would like to review it. However, I am coming off some time up so have a lot of PRs in the queue -- I will try to get to this asap in 1-2 days

**Review by howardjohn — CHANGES_REQUESTED** [review]
I don't think this is quite right. We document what to do, just poorly. I opened https://github.com/istio/istio.io/pull/13420 to make the docs more correct.

I also think we should probably make the default defaultRevision `default` (wow, what a confusing sentence!)

Thoughts @Monkeyanator @jacob-delgado ?

**Comment by zirain** [comment]
> I don't think this is quite right. We document what to do, just poorly. I opened [istio/istio.io#13420](https://github.com/istio/istio.io/pull/13420) to make the docs more correct.
> 
> I also think we should probably make the default defaultRevision `default` (wow, what a confusing sentence!)
> 
> Thoughts @Monkeyanator @jacob-delgado ?

did I miss something? this's not things about revision, it's about `configValidation`(webhook).

**Comment by howardjohn** [comment]
webhook is configured properly as long as you set default revision

**Comment by zirain** [comment]
here's my question: why there's another webhook here: https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml

**Comment by hanxiaop** [comment]
> here's my question: why there's another webhook here: https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml

`istio-validator-xxx` is more like a webhook for a specific revision, served per-revision basis. If you install a control-plane with revision, this will be installed. `istio-default-validator` is more like a default webhook for resources.

File: manifests/charts/istio-control/istio-discovery/values.yaml (hunk 1 above)
Comment (jacob-delgado) [reply manifests/charts/istio-control/istio-discovery/values.yaml:221]: We'd need to do this for istioctl based installs as well. Maybe we remove this completely? I don't know how this is used

**Comment by jacob-delgado** [comment]
So https://github.com/istio/istio/blob/5ddb5c94c424eeafb7040e0d21008c7aee42dc82/manifests/profiles/default.yaml#L92 is set when using the default profile for istioctl. This makes sense for us to also set for the helm chart since it's not set and is false by default. I just don't know when you'd want to set `configValidation` to `false`. It may be better for us to never allow users to set this, but I'd have to get @howardjohn to comment on this.

The PR changed since it was originally put up. Can you update the original description @ichbinfrog ?

**Pull request closed without merging**